pandas/core/internals/blocks.py

```diff
@@ -1471,7 +1471,15 @@ def fillna(
                 # We support filling a DatetimeTZ with a `value` whose timezone
                 #  is different by coercing to object.
                 if self.dtype.kind == "m":
-                    # TODO: don't special-case td64
+                    # GH#45746
+                    warnings.warn(
+                        "The behavior of fillna with timedelta64[ns] dtype and "
```

Comment:
can you add `{type(value)}` here to show what the user passed?

Reply:
updated + green